App/iOS/Delegates/SceneDelegate.swift

```diff
@@ -200,6 +200,8 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     if Preferences.URP.referralLookupOutstanding.value == false {
       appDelegate.dau.sendPingToServer()
     }
+    
+    BraveSkusManager.refreshSKUCredential(isPrivate: PrivateBrowsingManager.shared.isPrivateBrowsing)
```

Comment:
This call is done every time the app enters foreground. It does not need to be remove from memory.
Apple is pretty strict with killing background tasks and timers, I think checking every time the app is opened is good enough for our case.